### PR TITLE
added protostream decoder functionality

### PR DESCRIPTION
**Pull request by NeerajGartia21**

### CODE REVIEW

File: lib/protocols.js [hunk 1]
```diff
@@ -801,6 +801,21 @@
       }
     }
 
+    function decodeProtobufMessage(wrappedMessage){
+        var wrappedTypeId = wrappedMessage.wrappedTypeId;
+        var wrappedMessage= wrappedMessage.wrappedMessage;
+        var protostreamType = ProtostreamType.lookupProtostreamTypeById(wrappedTypeId);
+        var root = ProtobufRoot.findRootByTypeName(protostreamType.protostreamTypeName);
+        var userProtobufType= root.lookupType(protostreamType.protostreamTypeName);
```
Comment (rigazilla) [inline lib/protocols.js:809]: `lookupType` is called twice in these lines. Decoder is executed for each incoming message, better to try to optimize the code. See my comments below

File: lib/protocols.js [hunk 2]
```diff
@@ -1039,6 +1054,24 @@
     }
   }());
 
+  var ProtobufRoot = (function() {
+    var protobufRoots = [];
```
Comment (rigazilla) [inline lib/protocols.js:1058]: we're offering the possibility to register multiple roots, but this isn't a requirement. Does this add any advantage? How about having just one to speed up decoding and simplify code?

Comment (NeerajGartia21) [reply lib/protocols.js:1058]: Yeah, Its not a requirement. I will remove this.

**Comment by rigazilla** [comment]
I've added a simple get test. Works like a charm!

**Comment by NeerajGartia21** [comment]
Thanks sir!

**Comment by rigazilla** [comment]
merged, thanks @NeerajGartia21 !

**Pull request merged**